src/functions/OverloadedOoops.kts

```diff
@@ -0,0 +1,29 @@
+package functions.overloadedOoops
+
+class Test {
+
+    fun test(block: () -> Int = { 0 }): String {
+        return "${block()}"
```

Comment:
thoughts on `block().toString()`?

Reply:
Yeah, I think it's much simpler: it reads as "call the block" and the "convert result to string". The current solution has a lot of symbols: `"${()}"`, which is tricky to parse, even with highlight.

Reply:
I think it's worth simplifying as that's not the crux of the puzzle, I don't think this puzzle needs a red herring, it's complex enough as it is :)

Reply:
@TWiStErRob , don't you mind if I replace previous variant with that one (described above)? I think it looks simpler, less code, but demonstrates exactly the same problem as initial code.